src/v/cloud_storage_clients/abs_error.cc

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2022 Redpanda Data, Inc.
+ *
+ * Licensed as a Redpanda Enterprise file under the Redpanda Community
+ * License (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * https://github.com/redpanda-data/redpanda/blob/master/licenses/rcl.md
+ */
+
+#include "cloud_storage_clients/abs_error.h"
+
+#include <boost/lexical_cast.hpp>
+
+#include <map>
+
+namespace cloud_storage_clients {
+
+// NOLINTNEXTLINE
```

Comment:
Note for reviewer: see docs for error codes below:
* https://learn.microsoft.com/en-us/rest/api/storageservices/common-rest-api-error-codes
* https://learn.microsoft.com/en-us/rest/api/storageservices/blob-service-error-codes